src/overlays/actors/ovl_En_Zl2/z_en_zl2.c

```diff
@@ -27,265 +101,1664 @@ const ActorInit En_Zl2_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51C64.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51CA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51D0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51D24.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51DA4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51EA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51EBC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51F38.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B51FA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B52068.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B52098.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B52108.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B52114.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B521A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/EnZl2_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/EnZl2_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B52348.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B523BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B523C8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/func_80B525D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl2/EnZl2_Draw.s")
+#endif
+
+s32 func_80B4F45C(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                  Gfx** gfx) {
+    s32 pad;
+    EnZl2* this = THIS;
+    Mtx* sp74;
+    MtxF sp34;
+    Vec3s sp2C;
+    s16 pad2;
+    s16* unk_1DC = this->unk_1DC;
+
+    if (limbIndex == 0xE) {
+        sp74 = Graph_Alloc(globalCtx->state.gfxCtx, sizeof(Mtx) * 7);
+        gSPSegment(gfx[0]++, 0x0C, sp74);
+
+        Matrix_Push();
+        Matrix_Translate(pos->x, pos->y, pos->z, MTXMODE_APPLY);
+        Matrix_RotateRPY(rot->x, rot->y, rot->z, MTXMODE_APPLY);
+        Matrix_Push();
+        Matrix_Translate(362.0f, -133.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_Get(&sp34);
+        func_800D20CC(&sp34, &sp2C, 0);
+        if (func_800C0D28(globalCtx) == 0) {
+            func_80B4EE38(this, sp2C.y, 0);
+            func_80B4F230(this, sp2C.x, 1);
+            func_80B4EF64(this, sp2C.z, 2);
+        }
+        Matrix_RotateRPY(unk_1DC[0] + kREG(31), unk_1DC[1] + kREG(32), unk_1DC[2] + kREG(33), MTXMODE_APPLY);
+        Matrix_Translate(-188.0f, -184.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_ToMtx(sp74, "../z_en_zl2.c", 1056);
```

Comment:
Since sp74 is basically a pointer to an array of matrices, I suggest using array accesses instead pointer arithmetic.
```suggestion
        Matrix_ToMtx(&sp74[0], "../z_en_zl2.c", 1056);
```